pulpcore/app/serializers/base.py

```diff
@@ -122,6 +122,55 @@ def __init_subclass__(cls, **kwargs):
         except AttributeError:
             pass
 
+    def _update_labels(self, instance, labels):
+        """
+        Update the labels for a Model instance.
+
+        Args:
+            instance (pulpcore.app.models.BaseModel): instance with labels to update
+            labels (list): labels to set for the instance
+        """
+        instance.pulp_labels.exclude(key__in=labels.keys()).delete()
+
+        for key, value in labels.items():
+            label = instance.pulp_labels.filter(key=key).first()
```

Comment:
Is this the same as
```suggestion
            label = instance.pulp_labels.get(key=key)
```
because there can never be more than one?

Reply:
Yea, there should only be one. Although `get()` raises an exception if there is no match where as `first()` returns None. I could rewrite this code with `get()` though:

```python
for key, value in labels.items():
    try:
        label = instance.pulp_labels.get(key=key)
        if label.value != value:
            instance.pulp_labels.filter(key=key).update(value=value)
    except Label.DoesNotExist:
        instance.pulp_labels.create(key=key, value=value)
```

Reply:
Nah, i think it's ok. Thanks for the clarification.